python/ray/serve/tests/test_api.py

```diff
@@ -251,17 +252,14 @@ def __call__(self, flask_request, temp=None):
     serve.create_backend(
         "bsimple:v1",
         BatchSimple,
-        config={
-            "max_batch_size": 2,
-            "num_replicas": 3
-        })
+        config=BackendConfig(max_batch_size=2, num_replicas=3))
```

Comment:
can we add a test for legacy behavior? the case where config is a dictionary in create_backend and update_backend_config